backend/retentionpolicy.ts

```diff
@@ -0,0 +1,180 @@
+import { model } from 'mongoose'
+import {
+  AnyState,
+  ExpenseReport as IExpenseReport,
+  HealthCareCost as IHealthCareCost,
+  Travel as ITravel,
+  User as IUser,
+  Locale,
+  ReportType,
+  reportIsHealthCareCost,
+  reportIsTravel,
+  schemaNames
+} from '../common/types.js'
+import i18n from './i18n.js'
+import { sendMail } from './mail/mail.js'
+import Settings from './models/settings.js'
+import User from './models/user.js'
+
+async function getForRetentionPolicy(schema: schemaNames, date: Date, state: AnyState) {
+  return await model(schema).find({ state: state, updatedAt: { $lt: date }, historic: false })
```

Comment:
Hier kannst du gerne noch .lean() benutzen macht die Abfrage um einiges schneller (Kannst du in der mongoose Dokumentation nachschauen.


```suggestion
  return await model(schema).find({ state: state, updatedAt: { $lt: date }, historic: false }).lean()
```

Reply:
mit dem lean() bekomme ich einen Datentypen Fehler bei den reports in der Funktion notificationMailForDeletions() und triggerDeletion()

Reply:
Wahrscheinlich weil die Funktion model('...').find() den Typen any zurück gibt und mit .lean() jetzt etwas mehr eingeschränkteres.

Versuch doch den TypeError zu lösen.